src/neo/SmartContract/Native/ManagementContract.cs

```diff
@@ -0,0 +1,132 @@
+#pragma warning disable IDE0051
+
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using System;
+using System.Numerics;
+
+namespace Neo.SmartContract.Native
+{
+    public sealed class ManagementContract : NativeContract
+    {
+        public override string Name => "Neo Contract Management";
+        public override int Id => 0;
+        public override uint ActiveBlockIndex => 0;
+
+        private const byte Prefix_NextAvailableId = 15;
+
+        private int GetNextAvailableId(StoreView snapshot)
+        {
+            StorageItem item = snapshot.Storages.GetAndChange(new KeyBuilder(Id, Prefix_NextAvailableId), () => new StorageItem(1));
+            int value = (int)(BigInteger)item;
+            item.Add(1);
```

Comment:
`item.Add(1);` should be before `int value = (int)(BigInteger)item;` or it is a reference?

Reply:
We need to get the old value.